view/template/layout/_analytics_footer.php

```diff
@@ -8,13 +8,16 @@
 
     ga('create', 'UA-60403362-1', 'auto');
     ga('require', 'GTM-NT8579P');
+    ga('set', 'userId', '<?php echo $_SESSION[Session::USER_ID] ?>');
     ga('send', 'pageview');
 
    !function(f,b,e,v,n,t,s) {if(f.fbq)return;n=f.fbq=function(){n.callMethod? n.callMethod.apply(n,arguments):n.queue.push(arguments)}; if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0'; n.queue=[];t=b.createElement(e);t.async=!0; t.src=v;s=b.getElementsByTagName(e)[0]; s.parentNode.insertBefore(t,s)}(window,document,'script', 'https://connect.facebook.net/en_US/fbevents.js');
 
-   fbq('init', '1618717031725766');
+   fbq('init', '1618717031725766',{uid: '<?php echo $_SESSION[Session::USER_ID] ?>'}));
    fbq('track', '<?php echo Response::getFacebookPixelAnalyticsType() ?>');
 
+    <?php LBRY::logWebVisitor('PHP', $_SESSION[Session::USER_ID], $_SERVER['REMOTE_ADDR'])?>
```

Comment:
This technically breaks MVC, as a template shouldn't be initiating remote calls (unless it's doing so client-side).

There's not a super-obvious place to put this, I'd say either part of `Session::init` or in `Controller::execute` or `Controller::dispatch`.

Reply:
(An alternative design is to create one of an/or both `Actions::preExecute` and `Actions::postExecute`, to be called on both sides of `Controller::execute`)